index.js

```diff
@@ -128,52 +154,50 @@ class S3Adapter {
       const serializedTags = serialize(options.tags);
       params.Tagging = serializedTags;
     }
-    return this.createBucket().then(() => new Promise((resolve, reject) => {
-      this._s3Client.upload(params, (err, response) => {
-        if (err !== null) {
-          return reject(err);
-        }
-        return resolve(response);
-      });
-    }));
+    await this.createBucket();
+    const command = new PutObjectCommand(params);
+    const response = await this._s3Client.send(command);
+    const location = `https://${this._bucket}.s3.${this._region}.amazonaws.com/${params.Key}`;
```

Comment:
In Parse Server's FileRouter (line 237), the url and name are used right after file creation, so we need a valid URL immediately. Since the S3 SDK doesn’t provide a URL after creating the file, I manually compose one using the bucket and key.

That said, I could update this to better support external endpoints, making it more flexible.

Reply:
I added this part, could you please review it?